profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallbackFactory.cpp

```diff
@@ -70,6 +70,17 @@ HRESULT STDMETHODCALLTYPE CorProfilerCallbackFactory::CreateInstance(IUnknown* p
         return CLASS_E_NOAGGREGATION;
     }
 
+    // the scenario where different CLRs are loaded in the same process is not supported
```

Comment:
cannot we be in a race and both call to `CreateInstance` has not yet call the creation of `CorProfilerCallback` ?
In that case, we could end up with the same issue.
Having an an `atomic<bool>` would fix it no? (or a mutex in some extent)